packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -219,10 +219,12 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
       this.certificateArns.push(first.certificateArn);
     }
 
-    if (additionalCerts.length > 0) {
+    // Only one certificate can be specified per resource, even though 
+    // `certificates` is of type Array
+    for (const cert of additionalCerts) {
       new ApplicationListenerCertificate(this, id, {
```

Comment:
The `id` parameter will need to change; we can't have multiple constructs with the same id in the same scope. Should be as simple as appending a counter to the id, although for backwards-compatibility reasons, we would probably only want to append the id to the 2+ object. (e.g., 'id', 'id2', 'id3').